code/modules/atmospherics/machinery/components/components_base.dm

```diff
@@ -31,7 +31,7 @@
 	underlays.Cut()
 
 	var/turf/T = loc
-	if(level == 2 || !T.intact)
+	if(level == 2 || istype(T) && !T.intact)
```

Comment:
Pipe building uses temporary pipes in nullspace to get the connection directions. Very ugly, but there's no simple alternative.